src/index.ts

```diff
@@ -134,6 +134,130 @@ app.post("/api/users", async (req, res) => {
   res.json(user);
 });
 
+/**
+ * @openapi
+ * /api/users/{id}:
+ *   patch:
+ *      summary: Update particular data for a user
+ *      description: Ur
+ *      parameters:
+ *        - name: id
+ *          in: path
+ *          description: User id to update
+ *          required: true
+ *          schema:
+ *            type: integer
+ *      requestBody:
+ *        description: Update customer with properties to be changed
+ *        content:
+ *         application/json:
+ *           schema:
+ *             type: object
+ *             name: user data
+ *             required: true
+ *             properties:
+ *               name:
+ *                 type: string
+ *                 description: User name
+ *                 example: John Doe
+ *               email:
+ *                 type: string
+ *                 description: User email
+ *                 example: test@test.pl
+ *      responses:
+ *        200:
+ *          description: Successfully updated a user
+ *          content:
+ *            application/json:
+ *              schema:
+ *                type: object
+ *                properties:
+ *                  data:
+ *                      type: object
+ *                      properties:
+ *                        id:
+ *                          type: integer
+ *                          description: User id
+ *                          example: 1
+ *                        name:
+ *                          type: string
+ *                          description: Username
+ *                        email:
+ *                          type: string
+ *                          description: Username
+ *        400:
+ *          description: Missing id of a user
+ */
+
+app.patch("/api/users/:id", async (req, res) => {
+  const data = req.body;
+  const { id } = req.params;
+
+  if (!id) {
+    return res
+      .status(400)
+      .json({ error: "Missing id of user to update" });
+  }
+
+  try {
+    const user = await prisma.user.update({
+      where: {
+        id: Number(id)
+      },
+      data: {
+        name: data.name || undefined,
+        email: data.email || undefined, 
+      }
+    })
+
+    res.json(user);
+  } catch (e) {
+    res.status(404).json({});
```

Comment:
That's enough for our example, but just so you know - there are different error types that can appear here and with your code I'm able to get to this catch block not only by specifying incorrect ID, but also when I send e.g. 
```json
{
  "email": 1
}
```

as the body. I'll just get [PrismaClientValidationError](https://www.prisma.io/docs/reference/api-reference/error-reference#prismaclientvalidationerror) that says that `email` should be a string and not an integer. But don't worry, we'll cover this case when working on prisma in details later on - this is a good start!